std/range/package.d

```diff
@@ -4125,32 +4125,60 @@ pure unittest
     Generate lockstep's opApply function as a mixin string.
     If withIndex is true prepend a size_t index to the delegate.
 */
-private string lockstepMixin(Ranges...)(bool withIndex)
+private string lockstepMixin(Ranges...)(bool withIndex, bool reverse)
 {
     import std.format : format;
 
     string[] params;
     string[] emptyChecks;
     string[] dgArgs;
     string[] popFronts;
+    string indexDef;
+    string indexInc;
 
     if (withIndex)
     {
         params ~= "size_t";
         dgArgs ~= "index";
+        if (reverse)
+        {
+            indexDef = q{
+                size_t index = ranges[0].length-1;
+                enforce(_stoppingPolicy == StoppingPolicy.requireSameLength, "lockstep can only be used with foreach_reverse when stoppingPolicy == requireSameLength");
```

Comment:
Uhm... we just agreed on the polar opposite of that. Supporting indexed reverse iteration with `shortest` will either:

- break symmetry with regular `foreach_reverse` (indexes would be wrong). Or

- incur a hidden cost to skip trailing elements.

I am not completely unwilling to violate the first of these, and very reluctant to break the latter.

For non-indexed `foreach_reverse`, `shortest` is already supported and behaves the way you want:

```
foreach_reverse (e1, e2; lockstep([1,2,3], [4,5])) {
    // Will iterate over (3,5) and (2,4) in that order.
}
```